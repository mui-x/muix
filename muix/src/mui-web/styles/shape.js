//----------------------------------------------------------------------------------
//
// This code was generated from material-ui v1.4.2 by reactxx-codemod tool
// (https://github.com/reactxx/reactxx/tree/master/codemod)
//
//----------------------------------------------------------------------------------

const shape = {
  borderRadius: 4
};
export default shape;
