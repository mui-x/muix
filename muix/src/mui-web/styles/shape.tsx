//autogenerated--------------------------------------------------------------------
// 
// This code was generated from material-ui v3.0.0 by reactxx-codemod tool
// (https://github.com/reactxx/reactxx/tree/master/codemod)
// 
//----------------------------------------------------------------------------------

export interface Shape {
  borderRadius: number;
}
export type ShapeOptions = Partial<Shape>;
const shape = {
  borderRadius: 4
};
export default shape;
