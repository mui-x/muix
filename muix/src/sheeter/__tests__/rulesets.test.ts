import {
  atomizeRuleset,
  $web, $native,
} from "reactxx-sheeter"
import {TSheeter} from "reactxx-typings"
import { Shape, theme } from "reactxx-typings-test/shape"
import { initPlatform } from "./init-platform"

const t = $web({ color: 'green' })

describe("ATOMIZE RULESET", () => {

  const doTest = (isWeb: boolean) => {
    beforeEach(() => initPlatform(isWeb))
    let ruleset

    it("01 (null)", () => {
      ruleset = atomizeRuleset(null)
      expect(ruleset).toMatchSnapshot()
    })

    it("02 ({})", () => {
      ruleset = atomizeRuleset({})
      expect(ruleset).toMatchSnapshot()
    })

    it("03 ([null, undefined, {}])", () => {
      ruleset = atomizeRuleset([null, undefined, {}])
      expect(ruleset).toMatchSnapshot()
    })

    it("04 ({ color: 'red', margin: 0 })", () => {
      ruleset = atomizeRuleset({ color: 'red', margin: 0 })
      expect(ruleset).toMatchSnapshot()
    })

    it("05 ([{ color: 'red' }, { margin: 0 }])", () => {
      ruleset = atomizeRuleset([{ color: 'red' }, { margin: 0 }])
      expect(ruleset).toMatchSnapshot()
    })

    it("06 (...$web: { color: 'green' }, $native: { color: 'blue' })", () => {
      ruleset = atomizeRuleset([
        {
          color: 'red',
        },
        $web({ color: 'green' }),
        $native({ color: 'blue' }),
      ])
      expect(ruleset).toMatchSnapshot()
    })

    it("07 atomizeRuleset(atomizeRuleset({color: 'red'", () => {
      ruleset = atomizeRuleset(atomizeRuleset({
        color: 'red'
      }))
      expect(ruleset).toMatchSnapshot()
    })

    it("08 atomizeRuleset({$web: atomizeRuleset({ color: 'red', $native: atomizeRuleset({ color: 'green'", () => {
      ruleset = atomizeRuleset([
        $web(atomizeRuleset({ color: 'red' })),
        $native(atomizeRuleset({ color: 'green' }))
      ])
      expect(ruleset).toMatchSnapshot()
    })

    it("09 ERROR for atomizeRuleset({:hover': atomizeRuleset(...", () => {
      const fnc = () => {
        window.isWeb && atomizeRuleset(
          $web({
            ':hover': atomizeRuleset({ color: 'red' }) as any,
          }
        ))
      }
      window.isWeb && expect(fnc).toThrow(/.*/)
    })

    it("10 ERROR for atomizeRuleset({:hover': [ARRAY]", () => {
      const fnc = () => window.isWeb && atomizeRuleset(
        $web({
          ':hover': [
            { color: 'red' },
            { margin: 10 }
          ] as any
        })
      )
      window.isWeb && expect(fnc).toThrow(/.*/)
    })

    it("11 with theme", () => {
      ruleset = atomizeRuleset<'root', Shape>(theme => [
        theme.primary.normal,
        $web(theme.secondary.normal),
      ], theme)
      expect(ruleset).toMatchSnapshot()
    })

    it("12 ATOMIZE modifies source object", () => {
      const source = {
        color: 'blue',
        $web: { color: 'red' },
        $native: { color: 'green' },
      }
      ruleset = atomizeRuleset(source)
      expect(ruleset).toMatchSnapshot()
      expect(source).toMatchSnapshot()
    })
  }

  describe("## NATIVE ##", () => doTest(false))
  describe("## WEB ##", () => doTest(true))

})

