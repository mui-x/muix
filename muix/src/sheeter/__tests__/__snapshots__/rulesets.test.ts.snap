// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ATOMIZE RULESET ## NATIVE ## 01 (null) 1`] = `null`;

exports[`ATOMIZE RULESET ## NATIVE ## 02 ({}) 1`] = `null`;

exports[`ATOMIZE RULESET ## NATIVE ## 03 ([null, undefined, {}]) 1`] = `null`;

exports[`ATOMIZE RULESET ## NATIVE ## 04 ({ color: 'red', margin: 0 }) 1`] = `
Array [
  Array [
    "color: red /*.*/",
    "margin: 0 /*.*/",
  ],
]
`;

exports[`ATOMIZE RULESET ## NATIVE ## 05 ([{ color: 'red' }, { margin: 0 }]) 1`] = `
Array [
  Array [
    "color: red /*.[0]*/",
  ],
  Array [
    "margin: 0 /*.[1]*/",
  ],
]
`;

exports[`ATOMIZE RULESET ## NATIVE ## 06 (...$web: { color: 'green' }, $native: { color: 'blue' }) 1`] = `
Array [
  Array [
    "color: red /*.*/",
  ],
  Array [
    "color: blue /*./$native*/",
  ],
]
`;

exports[`ATOMIZE RULESET ## NATIVE ## 07 atomizeRuleset(atomizeRuleset({color: 'red' 1`] = `
Array [
  Array [
    "color: red /*.*/",
  ],
]
`;

exports[`ATOMIZE RULESET ## NATIVE ## 08 atomizeRuleset({$web: atomizeRuleset({ color: 'red', $native: atomizeRuleset({ color: 'green' 1`] = `
Array [
  Array [
    "color: green /*.*/",
  ],
]
`;

exports[`ATOMIZE RULESET ## NATIVE ## 11 with theme 1`] = `
Array [
  Array [
    "color: white /*.[0]*/",
    "backgroundColor: darkgreen /*.[0]*/",
  ],
]
`;

exports[`ATOMIZE RULESET ## NATIVE ## 12 ATOMIZE modifies source object 1`] = `
Array [
  Array [
    "color: blue /*.*/",
  ],
  Array [
    "color: green /*./$native*/",
  ],
]
`;

exports[`ATOMIZE RULESET ## NATIVE ## 12 ATOMIZE modifies source object 2`] = `
Object {
  "color": "blue",
}
`;

exports[`ATOMIZE RULESET ## WEB ## 01 (null) 1`] = `null`;

exports[`ATOMIZE RULESET ## WEB ## 02 ({}) 1`] = `null`;

exports[`ATOMIZE RULESET ## WEB ## 03 ([null, undefined, {}]) 1`] = `null`;

exports[`ATOMIZE RULESET ## WEB ## 04 ({ color: 'red', margin: 0 }) 1`] = `
Array [
  Array [
    ".a { color:red /*.*/ }",
    ".b { margin:0 /*.*/ }",
  ],
]
`;

exports[`ATOMIZE RULESET ## WEB ## 05 ([{ color: 'red' }, { margin: 0 }]) 1`] = `
Array [
  Array [
    ".a { color:red /*.[0]*/ }",
  ],
  Array [
    ".b { margin:0 /*.[1]*/ }",
  ],
]
`;

exports[`ATOMIZE RULESET ## WEB ## 06 (...$web: { color: 'green' }, $native: { color: 'blue' }) 1`] = `
Array [
  Array [
    ".a { color:red /*.*/ }",
  ],
  Array [
    ".b { color:green /*./$web*/ }",
  ],
]
`;

exports[`ATOMIZE RULESET ## WEB ## 07 atomizeRuleset(atomizeRuleset({color: 'red' 1`] = `
Array [
  Array [
    ".a { color:red /*.*/ }",
  ],
]
`;

exports[`ATOMIZE RULESET ## WEB ## 08 atomizeRuleset({$web: atomizeRuleset({ color: 'red', $native: atomizeRuleset({ color: 'green' 1`] = `
Array [
  Array [
    ".a { color:red /*.*/ }",
  ],
]
`;

exports[`ATOMIZE RULESET ## WEB ## 11 with theme 1`] = `
Array [
  Array [
    ".a { color:white /*.[0]*/ }",
    ".b { background-color:darkgreen /*.[0]*/ }",
  ],
  Array [
    ".c { color:black /*.[1]/$web*/ }",
    ".d { background-color:lightblue /*.[1]/$web*/ }",
  ],
]
`;

exports[`ATOMIZE RULESET ## WEB ## 12 ATOMIZE modifies source object 1`] = `
Array [
  Array [
    ".a { color:blue /*.*/ }",
  ],
  Array [
    ".b { color:red /*./$web*/ }",
  ],
]
`;

exports[`ATOMIZE RULESET ## WEB ## 12 ATOMIZE modifies source object 2`] = `
Object {
  "color": "blue",
}
`;
